devicetypes/zooz/zooz-zse44-temperature-humidity-xs-sensor.src/zooz-zse44-temperature-humidity-xs-sensor.groovy

```diff
@@ -0,0 +1,418 @@
+	logDebug "${cmd}"
+	switch (cmd.notificationType) {
+		case temperatureAlarm.notificationType:
+			sendAlarmEvent(temperatureAlarm, cmd)
+			break
+		case humidityAlarm.notificationType:
+			sendAlarmEvent(humidityAlarm, cmd)
+			break
+		default:
+			logDebug "${cmd}"
+	}
+}
+
+void sendAlarmEvent(Map alarm, cmd) {
+	String value
+	if ((cmd.event == alarm.highEvent) || (cmd.event == alarm.lowEvent)) {
+		value = (cmd.event == alarm.highEvent) ? "high" : "low"
+	} else if (cmd.event == alarm.normalEvent) {
+		value = "normal"
+	}
+
+	if (value) {
+		logDebug "${alarm.name} is ${value}"
+		sendEvent(name: alarm.name, value: value)
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	switch (cmd.sensorType) {
+		case temperatureSensor.sensorType:
+			def temperature = convertTemperatureIfNeeded(cmd.scaledSensorValue, (cmd.scale ? "F" : "C"), cmd.precision)
+			sendTemperatureEvent(temperature)
+			break
+		case humiditySensor.sensorType:
+			sendHumidityEvent(cmd.scaledSensorValue)
+			break
+		default:
+			logDebug "Unhandled: ${cmd}"
+	}
+}
+
+void sendTemperatureEvent(value) {
+	state.displayTemperature = "${value}°${temperatureScale}"
+	logDebug "temperature is ${value}°${temperatureScale}"
+	sendEvent(name: "temperature", value: value, unit: temperatureScale)
+	sendTemperatureHumidityEvent()
+}
+
+void sendHumidityEvent(value) {
+	state.displayHumidity = "${safeToInt(value)}%"
+	logDebug "humidity is ${value}%"
+	sendEvent(name: "humidity", value: value, unit: "%")
+	sendTemperatureHumidityEvent()
+}
+
+void sendTemperatureHumidityEvent() {
+	sendEvent(name: "temperatureHumidity", value: "${state.displayTemperature} | ${state.displayHumidity}", displayed: false)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
+	logDebug "${cmd}"
+	sendEvent(name: "firmwareVersion", value: (cmd.applicationVersion + (cmd.applicationSubVersion / 100)))
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	runIn(4, refreshSyncStatus)
+	String name = configParams.find { name, param -> param.num == cmd.parameterNumber }?.key
+	if (name) {
+		int val = cmd.scaledConfigurationValue
+
+		if ((val < 0) && ((name == "humidityOffset") || (name == "tempOffset"))) {
+			val = (val + 256)
+		}
+
+		state[name] = val
+		logDebug "${configParams[name]?.title}(#${configParams[name]?.num}) = ${val}"
+	} else {
+		logDebug "Parameter #${cmd.parameterNumber} = ${cmd.scaledConfigurationValue}"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.Command cmd) {
+	logDebug "Unhandled zwaveEvent: ${cmd}"
+}
+
+void refreshSyncStatus() {
+	int changes = pendingChanges
+	sendEvent(name: "syncStatus", value: (changes ?  "${changes} Pending Changes" : "Synced"), displayed: false)
+}
+
+Integer getPendingChanges() {
+	int configChanges = safeToInt(configParams.count { name, param ->
+		(getSettingVal(name) != getStoredVal(name))
+	}, 0)
```

Comment:
Could You reformat this block of code ?

Reply:
I'm not sure how I could make the format any cleaner, but I removed the safeToInt function wrapping it because it was redundant.

Out of curiosity, I had that identical code in the PR for the ZSE43, which you reviewed at the beginning of the month, so why wasn't the formatting an issue in that handler?